applications/HDF5Application/custom_io/hdf5_container_component_io.cpp

```diff
@@ -389,477 +96,331 @@ ContainerComponentIO<TContainerType, TContainerItemType, TComponents...>::Contai
-}
+    // Write block partition.
+    WritePartitionTable(*mpFile, mComponentPrefix, info);
 
-// Adding node other variable WriteRegisteredComponent template definition
-template <>
-template <typename... Targs>
-void ContainerComponentIO<NodesContainerType,
-                          NodeType,
-                          Variable<array_1d<double, 3>>,
-                          Variable<double>,
-                          Variable<int>,
-                          Variable<Vector<double>>,
-                          Variable<Matrix<double>>>::WriteRegisteredComponent(const std::string& rComponentName,
-                                                                              Targs&... args)
-{
-    RegisteredComponentLookup<Variable<array_1d<double, 3>>, Variable<double>, Variable<int>,
-                              Variable<Vector<double>>, Variable<Matrix<double>>>(rComponentName)
-        .Execute<WriteNodeComponent>(args...);
+    KRATOS_CATCH("");
 }
 
-// Adding node flags ReadRegisteredComponent template definition
-template <>
-template <typename... Targs>
-void ContainerComponentIO<NodesContainerType, NodeType, Flags>::ReadRegisteredComponent(
-    const std::string& rComponentName, Targs&... args)
+template <class TContainerType, class TContainerDataIO, class... TComponents>
+std::map<std::string, Parameters> ContainerComponentIO<TContainerType, TContainerDataIO, TComponents...>::Read(
+    TContainerType& rLocalContainer,
+    const TContainerDataIO& rContainerDataIO,
+    Communicator& rCommunicator)
 {
-    RegisteredComponentLookup<Flags>(rComponentName).Execute<ReadNodeComponent>(args...);
-}
+    KRATOS_TRY;
 
-// Adding node other variable ReadRegisteredComponent template definition
-template <>
-template <typename... Targs>
-void ContainerComponentIO<NodesContainerType,
-                          NodeType,
-                          Variable<array_1d<double, 3>>,
-                          Variable<double>,
-                          Variable<int>,
-                          Variable<Vector<double>>,
-                          Variable<Matrix<double>>>::ReadRegisteredComponent(const std::string& rComponentName,
-                                                                             Targs&... args)
-{
-    RegisteredComponentLookup<Variable<array_1d<double, 3>>, Variable<double>, Variable<int>,
-                              Variable<Vector<double>>, Variable<Matrix<double>>>(rComponentName)
-        .Execute<ReadNodeComponent>(args...);
-}
+    std::map<std::string, Parameters> attributes;
 
-template <>
-template <class ...TArguments>
-void ContainerComponentIO<VertexContainerType, Vertex, Flags>::WriteRegisteredComponent(
-    const std::string& rComponentName, TArguments&... rArguments)
-{
-    RegisteredComponentLookup<Flags>(rComponentName).Execute<WriteVertexComponent>(rArguments...);
-}
+    if (mComponentNames.size() == 0) {
+        return attributes;
+    }
```

Comment:
Hmm.... I would say, these `mComponentNames` should be consistent among ranks (I remember your use case), but woudn't it be the responsibility of the person who gives this as input to make them consistent across ranks before providing it?

Reply:
Ok, it is passed via `Kratos::Parameters`, so we don't have to make the ctor of `ContainerComponentIO` private since, user is suppose to pass the same `Kratos::Parameters` from all ranks.